sellercenter-product-port/src/main/java/com/gfg/sellercenter/product/port/infra/JsonHttpReader.java

```diff
@@ -8,16 +8,16 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
-import java.io.Reader;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 
 public class JsonHttpReader {
-    private String readAll(Reader rd) throws IOException {
+    private String readAll(BufferedReader reader) throws IOException {
         StringBuilder sb = new StringBuilder();
-        int cp;
-        while ((cp = rd.read()) != -1) {
-            sb.append((char) cp);
```

Comment:
Adding one character at a time consumes lot of memory and can lead to OOM exception